File: app/src/main/java/com/example/brijeshkum/mybaseproject/di/AppComponent.java
package com.example.brijeshkum.mybaseproject.di;

import com.example.brijeshkum.mybaseproject.MainActivity;

import javax.inject.Singleton;

import dagger.Component;

@Singleton
@Component(modules={AppModule.class, NetModule.class})
public interface AppComponent {
    void inject(MainActivity activity);
    // void inject(MyFragment fragment);
    // void inject(MyService service);
}

File: app/src/main/java/com/example/brijeshkum/mybaseproject/di/AppComponent.kt
package com.example.brijeshkum.mybaseproject.di

//import com.example.brijeshkum.mybaseproject.HomeActivity
import com.example.brijeshkum.mybaseproject.MainActivity
import dagger.Component
import javax.inject.Singleton

@Singleton
@Component(modules = [AppModule::class, NetModule::class])
interface AppComponent {
    //fun inject(activity: HomeActivity) // void inject(MyFragment fragment);
    fun inject(activity: MainActivity)
}